pilot/pkg/dns/dns.go

```diff
@@ -354,8 +355,23 @@ func generateAltHosts(hostname string, nameinfo *nds.NameTable_NameInfo, proxyNa
 // of registry, we will look it up in one of our tables, failing which we will return NXDOMAIN.
 func (table *LookupTable) lookupHost(qtype uint16, hostname string) ([]dns.RR, bool) {
 	var hostFound bool
-	if _, hostFound = table.allHosts[hostname]; !hostFound {
-		// this is not from our registry
+
+	question := host.Name(hostname)
+	wildcard := false
+	// First check if host exists.
+	_, hostFound = table.allHosts[hostname]
+	if !hostFound {
+		// Check matching hosts for wildcard hosts.
+		for h := range table.allHosts {
+			if host.Name(h).Matches(question) {
```

Comment:
I think this would match `a.b.example.com` to `*.example.com` which is wrong?

Reply:
I think in the case we are trying to solve, is not svc.svcns.<dnsdomain> should match "*.<dnsdomain>" - so it is valid from that perspective